### PR TITLE
Fix crash on timeout

### LINKED ISSUE #8: Crash on timeout
```
12:37:20.964 [error] GenServer :ueberauth_token_worker_oauth2_token_adapters_quiqup terminating
** (Protocol.UndefinedError) protocol String.Chars not implemented for #PID<0.1564.0>. This protocol is implemented for: Atom, BitString, Date, DateTime, Float, Integer, List, NaiveDateTime, Time, URI, Version, Version.Requirement
    (elixir) /usr/local/src/elixir/lib/elixir/lib/string/chars.ex:3: String.Chars.impl_for!/1
    (elixir) /usr/local/src/elixir/lib/elixir/lib/string/chars.ex:22: String.Chars.to_string/1
    (ueberauth_token) lib/ueberauth_token/worker.ex:191: UeberauthToken.Worker.worker_details/1
    (ueberauth_token) lib/ueberauth_token/worker.ex:85: UeberauthToken.Worker.terminate/2
    (stdlib) gen_server.erl:648: :gen_server.try_terminate/3
    (stdlib) gen_server.erl:833: :gen_server.terminate/10
    (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Last message: :timeout
12:37:20.964 [warning] lager_error_logger_h dropped 46 messages in the last second that exceeded the limit of 50 messages/sec

=CRASH REPORT==== 12-Sep-2018::12:37:20 ===
  crasher:
    initial call: Elixir.UeberauthToken.Worker:init/1
    pid: <0.1564.0>
    registered_name: ueberauth_token_worker_oauth2_token_adapters_quiqup
    exception error: #{'__exception__' => true,
                       '__struct__' => 'Elixir.Protocol.UndefinedError',
                       description => <<>>,protocol => 'Elixir.String.Chars',
                       value => <0.1564.0>}
      in function  'Elixir.String.Chars':'impl_for!'/1 (/usr/local/src/elixir/lib/elixir/lib/string/chars.ex, line 3)
      in call from 'Elixir.String.Chars':to_string/1 (/usr/local/src/elixir/lib/elixir/lib/string/chars.ex, line 22)
      in call from 'Elixir.UeberauthToken.Worker':worker_details/1 (lib/ueberauth_token/worker.ex, line 191)
      in call from 'Elixir.UeberauthToken.Worker':terminate/2 (lib/ueberauth_token/worker.ex, line 85)
      in call from gen_server:try_terminate/3 (gen_server.erl, line 648)
      in call from gen_server:terminate/10 (gen_server.erl, line 833)
    ancestors: ['Elixir.UeberauthToken.Supervisor',<0.1554.0>]
    message_queue_len: 0
    messages: []
    links: [<0.1555.0>]
    dictionary: []
    trap_exit: false
    status: running
    heap_size: 1598
    stack_size: 27
    reductions: 2094
  neighbours:

=SUPERVISOR REPORT==== 12-Sep-2018::12:37:20 ===
     Supervisor: {local,'Elixir.UeberauthToken.Supervisor'}
     Context:    child_terminated
     Reason:     {#{'__exception__' => true,
                    '__struct__' => 'Elixir.Protocol.UndefinedError',
                    description => <<>>,protocol => 'Elixir.String.Chars',
                    value => <0.1564.0>},
                  [{'Elixir.String.Chars','impl_for!',1,
                       [{file,
                            "/usr/local/src/elixir/lib/elixir/lib/string/chars.ex"},
                        {line,3}]},
                   {'Elixir.String.Chars',to_string,1,
                       [{file,
                            "/usr/local/src/elixir/lib/elixir/lib/string/chars.ex"},
                        {line,22}]},
                   {'Elixir.UeberauthToken.Worker',worker_details,1,
                       [{file,"lib/ueberauth_token/worker.ex"},{line,191}]},
                   {'Elixir.UeberauthToken.Worker',terminate,2,
                       [{file,"lib/ueberauth_token/worker.ex"},{line,85}]},
                   {gen_server,try_terminate,3,
                       [{file,"gen_server.erl"},{line,648}]},
                   {gen_server,terminate,10,
                       [{file,"gen_server.erl"},{line,833}]},
                   {proc_lib,init_p_do_apply,3,
                       [{file,"proc_lib.erl"},{line,247}]}]}
     Offender:   [{pid,<0.1564.0>},
                  {id,'Elixir.UeberauthToken.Worker'},
                  {mfargs,
                      {'Elixir.UeberauthToken.Worker',start_link,
                          [[{provider,
                                'Elixir.Oauth2Token.Adapters.Quiqup'}]]}},
                  {restart_type,permanent},
                  {shutdown,5000},
                  {child_type,worker}]

12:37:20.964 [error] gen_server ueberauth_token_worker_oauth2_token_adapters_quiqup terminated with reason: #{'__exception__' => true,'__struct__' => 'Elixir.Protocol.UndefinedError',description => <<>>,protocol => 'Elixir.String.Chars',value => <0.1564.0>} in 'Elixir.String.Chars':'impl_for!'/1 line 3
12:37:20.964 [error] CRASH REPORT Process ueberauth_token_worker_oauth2_token_adapters_quiqup with 0 neighbours crashed with reason: #{'__exception__' => true,'__struct__' => 'Elixir.Protocol.UndefinedError',description => <<>>,protocol => 'Elixir.String.Chars',value => <0.1564.0>} in 'Elixir.String.Chars':'impl_for!'/1 line 3
12:37:20.965 [error] Supervisor 'Elixir.UeberauthToken.Supervisor' had child 'Elixir.UeberauthToken.Worker' started with 'Elixir.UeberauthToken.Worker':start_link([{provider,'Elixir.Oauth2Token.Adapters.Quiqup'}]) at <0.1564.0> exit with reason #{'__exception__' => true,'__struct__' => 'Elixir.Protocol.UndefinedError',description => <<>>,protocol => 'Elixir.String.Chars',value => <0.1564.0>} in 'Elixir.String.Chars':'impl_for!'/1 line 3 in context child_terminated
```

### PR DESCRIPTION
**Pull request by IanVaughan**
Escape strings correctly to protect against `protocol String.Chars not implemented` error.

Fixes #8

### CODE REVIEW

File: lib/ueberauth_token/worker.ex [hunk 1]
```diff
@@ -188,14 +191,17 @@ defmodule UeberauthToken.Worker do
...
-    line: #{env.line}
+    module: #{inspect(env.module)},
+    function: #{inspect(env.function)},
+    line: #{inspect(env.line)}
     """
   end
 
-  defp worker_details(provider) do
-    "worker #{worker_name(provider)} with process_id #{Process.whereis(worker_name(provider))}"
+  @spec worker_details(atom() | binary()) :: <<_::64, _::_*8>>
```
Comment (luizvarela) [inline lib/ueberauth_token/worker.ex:200]: can it be a `:: String.t` in the @spec?

**Comment by luizvarela** [comment]
LGTM

**Comment by IanVaughan** [comment]
Closing to reopen onto master, as GitHub doesnt let you edit the MR!

File: lib/ueberauth_token/worker.ex (hunk 1 above)
Comment (IanVaughan) [reply lib/ueberauth_token/worker.ex:200]: Yes, much nicer, done.

File: config/config.exs [hunk 2]
```diff
@@ -1,6 +1,6 @@
 use Mix.Config
 
-if Mix.env() == :test do
+if Mix.env() == :test || Mix.env() == :dev do
```
Comment (stephenmoloney) [inline config/config.exs:3]: Setting the config like this in `:dev` is something a developer would probably not expect.

**Pull request closed without merging**